### PR TITLE
POC : Docker implementation

### PR DESCRIPTION
**Pull request by gillesdubois**
Hi,

Here is the PR about the docker implementation i made some weeks ago. This one has been adapted to be usable in most of the configurations / installation.

This image is separated in 2 build :
- GoLang build image (latest ?)
- Alpine running image (3.9.4)

This will remove all GoLang build related deps / images to lower image size and make to more efficiant.

Question : I took your  base docker file to implement go compilation but why is golang image using latest tag ? From my experience its not really recommmanded to use latest tag on docker. Do you think we could set a tag to put a static golang image ? 

In this PR 4 files have been modified / added :

docker_config.yml :
* Used by the image to populate configuration using envsubst

Dockerfile :
* First layer is the go lang build, the only change i made is about the "ADD" instructions i replaced by "COPY". Using ADD is not really uesful when preforming a simple copy.
* Second layer will copy files from the first layer, define default env vars, install required deps, init configuration and run main
* Apart from 8080 there is no default expose concerning ACC servers, it will be indicated in the documentation

Entrypoint :
* From env vars, it will generate config file based on config_docker.yml file and start accserver

Question : it could be nice to provide an option to use your official releases on github. In our case, if someone or docker hub build an image at a time master branch is not working for X reason it might provoke bugs, frustrate people.

docker-compose.yml :
* Version 3 classic docker compose

So far, its running good for me but could you test it on your own ?

If this PR look good to you and is ready to merge i will write the documentation. I prefer to be sure everything is fine before writing doc :D 

Regards,
Gilles Dubois.

EDIT : Typos

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,39 +1,47 @@
 FROM golang AS build
-ADD . /go/src/github.com/assetto-corsa-web/accweb
+
+LABEL maintainer="Kugel" \
+	  contributors="GillesDubois" \
+      version="latest" \
+      description="Kugel's accweb app inside docker !"
```
Comment (Kugelschieber) [inline Dockerfile:6]: Probably change this to "Assetto Corsa Competizione Server Management Tool via Web Interface.". I don't need to have my name in it :)

**Comment by Kugelschieber** [comment]
Hey, thanks for all the work you put into this! 

```
Question : I took your base docker file to implement go compilation but why is golang image using latest tag ? From my experience its not really recommmanded to use latest tag on docker. Do you think we could set a tag to put a static golang image ?
```
Usually it's recommended to pin a version to prevent the build from breaking due to changes, but it's common to almost immediately switch to the latest Go version so the dockerfile uses that. I update Go on my machine as soon as a new version is released and never had a problem.

```
Question : it could be nice to provide an option to use your official releases on github. In our case, if someone or docker hub build an image at a time master branch is not working for X reason it might provoke bugs, frustrate people.
```
I've been building images manually in the past, but it's possible to build a tag too. So that shouldn't break.

One question: do we need a different configuration file? Can't we just use the existing sample configuration or include it using a volume?

**Comment by gillesdubois** [comment]
Hi @Kugelschieber ,

I will change the Description of the image this evening :)

There is no particular reason of using a dedicated config file. My first implementation used a dedicated config file and i didn't though of changing it.

The main advantage is that its controlled by the docker environement and ENV vars override is easier this way. There is no replace / config management everything is provided using the ENV vars.
Most of the docker images i built also have specific parameters dedicated to the environement (Kubernetes, etc..) so i always set a separate file from the original software. Do you want me to check we could change this ?

Regards,
Gilles Dubois.

**Comment by Kugelschieber** [comment]
Ah okay I haven't seen the variables inside the config file. Stupid me. That looks really good, so I'm okay with the PR after you've added the documentation.

**Comment by gillesdubois** [comment]
Ok no problems, i will write and make the minors changes this evening :)

**Comment by gillesdubois** [comment]
Hi @Kugelschieber ,

The documentation is ready, i'll let you review it :)

Regards,
Gilles Dubois.

**Comment by Kugelschieber** [comment]
Look good, thank you very much! I might make some adjustments to the readme, but it's ok for now.

**Pull request merged**